wolfcrypt/src/asn.c

```diff
@@ -9117,64 +9117,54 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
                           ecc_key* key, word32 inSz)
 {
     int    length;
-    int    ret = 0;
+    byte   b;
 
     if (input == NULL || inOutIdx == NULL || key == NULL || inSz == 0)
         return BAD_FUNC_ARG;
 
     if (GetSequence(input, inOutIdx, &length, inSz) < 0)
         return ASN_PARSE_E;
 
-#if defined(OPENSSL_EXTRA) || defined(ECC_DECODE_EXTRA)
-    {
-        byte b = input[*inOutIdx];
-        if (b != ASN_INTEGER) {
-            /* not from decoded cert, will have algo id, skip past */
-            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
-                return ASN_PARSE_E;
+    if (GetSequence(input, inOutIdx, &length, inSz) < 0)
```

Comment:
The alternate encoding call path doesn't work.
Assuming that there is a sequence followed by an integer, then wc_ecc_import_x963_ex is called.
The function checks the first byte for 4, 2 or 3 - point format. That is the import function assumes the input buffer is the point data and not the ASN.1 encoding.
Do we have test data for the OpenSSL encoding? Do you want me to change the code back and take the ASN.1 integer encoding off before calling import?